components/gfx/paint_task.rs

```diff
@@ -754,11 +623,59 @@ impl WorkerThread {
             content_age: tile.content_age,
         }
     }
+
+    fn create_layer_buffer_for_cpu_painted_tile(&mut self,
+                                                tile: BufferRequest,
+                                                draw_target: DrawTarget,
+                                                scale: f32)
+                                                -> Box<LayerBuffer> {
+        let mut layer_buffer = match tile.layer_buffer {
```

Comment:
nit: if you use `tile.layer_buffer.unwrap_or_else(|| { ... })` you can save a level of indentation